### DISCUSSION TITLE
Turbulence modeling: SST(-1994) vs SST-2003

### DISCUSSION THREAD

**Discussion by TobiKattmann** [su2code/SU2#1403] [Ideas]
Hi all,

For the work of @mheimgartner on species mixing (see #1332 ), some comparisons with Ansys Fluent were made. 

We noticed some differences in the results for axisymmetric flow with the SST turb model. After some digging of Mark the cause was found to be that Fluent uses the updated SST-2003 (https://www.afs.enea.it/project/neptunius/docs/fluent/html/th/node67.htm) model whereas SU2 implements the original SST version from 1994. (See https://turbmodels.larc.nasa.gov/sst.html for some info, the 1994-version is just the `SST` and the 2003-version is `SST-2003`

When hacking in the differences we had a much better agreement between the 2 codes. There is still some more thorough testing necessary whether everything is identical.

I am currently thinking of adding `KIND_TURB_MODEL= SST-2003` with the necessary changes (which are not too numerous, but tedious to fully get right.)

Why was the 1994 version implemented in the first place? Is someone more knowledgable about why prefer one over the other (I will read a bit into the available references but still some insight here would be helpful)

Best Tobi and Mark

@vdweide This is mostly fixing the discrepancies that were seen in the simulations 👍 
@bigfooted We have to think about how to go on with this, I would love to have some TUI files in version control for these things (planar/ axisymmetric, with/without Turb) to use that as a verification/plausibilisation approach 
@FlorianDm Maybe you are interested in this as well, we were looking at #1195 quite a bit for this work

**Comment by FlorianDm** [comment]
Hi, if you have any questions about my work I am happy to answer but I don't have any time to contribute or look at the code or anything at the moment. So this 1994/2003 discrepancy concerns the model in general I guess and not only the axisymmetry source terms. I have to admit I was just not aware of the newer version and I believe I often saw people citing Menter's original paper when referring to implementations in modern codes

**Comment by bigfooted** [comment]
My proposal is to first rewrite the current SST implementation, such that the implementation resembles the standard notation for the SST equations. It will then be easier to check for correctness. For instance for the transport equation of k, we have (from the larc website) 
`Drho*k/Dt = P - beta*rho*w*k + ...,` which should recognizable in su2 in this form imo.
Not sure if we should introduce a new turbulence model for this or work with additional keywords: 
`KIND_TURB_MODEL=SST `
`SST_MODEL=SST-V2003`
Note that there are 16 (!) SST versions just on the larc website, and you can mix versions, for instance SST-RC-Hellsten with SST-2003 to get SST-2003-RC-Hellsten. 
So you should be able to do:
`SST-V2003=TRUE`
`SST-RC_HELLSTEN=TRUE`

**Comment by pcarruscag** [comment]
What @suargi proposed should work for this #1364

**Comment by TobiKattmann** [comment]
The necessary changes of SST-1994 for SST-2003 can be found following what I wrote in  https://github.com/su2code/SU2/pull/1332#issuecomment-948314301 . That is not meant to be the final code, but this is a version @mheimgartner tested
